README.md

```diff
@@ -28,124 +54,44 @@ from float8_experimental.float8_linear_utils import (
+m = torch.compile(m)
+
 # toy training loop
 for _ in range(N_ITER):
     optimizer.zero_grad()
     y = m(x)
     y.sum().backward()
 
-    # specific to float8: separate step to sync scales/amaxes
+    # specific to float8 with delayed scaling: separate step to sync scales/amaxes
     # in the future, this may move to a context manager
     sync_float8_amax_and_scale_history(model)
 
     optimizer.step()
 ```
 
-## multi GPU
-
-```python
-from float8_experimental.tp_linear import swap_tp_linear_with_float8_linear
-
-# swaps the fairscale `ColumnParallelLinear` with `Float8ColumnParallelLinear`,
-# and the fairscale `RowParallelLinear` with `Float8RowParallelLinear`
-swap_tp_linear_with_float8_linear(model)
-
-# if applicable, enable sequence parallel on the right modules
-# TODO make the API for this nicer
-model.foo.bar.fc1.sequence_parallel = True
-model.foo.bar.fc2.sequence_parallel = True
-
-# the rest of the flow is the same as the single GPU flow
-```
-
-# high level technical design
-
-## UX
-
-We are using a module swap UX to keep things simple. If the user model has `torch.nn.Linear` modules or their `fairscale` TP/SP equivalents,
-we can convert them to float8. `F.linear`, `torch.mm`, `torch.matmul` are not supported at the moment.
+# upcoming work
 
-User is responsible for calling the `sync_float8_amax_and_scale_history` function once per fw/bw,
-this function updates the amax history. If distributed is enabled, this function also syncs amax values across workers.
-This is a separate model level function (as opposed to each module owning the syncing of its buffers) to
-make it easier to optimize performance (for example, reduce all the amaxes once in a single tensor instead of doing N reductions).
-
-Composability with `DTensor` is on our radar and we plan to look into this after the manual flow works e2e.
-
-A user facing tensor subclass UX is not being considered at the moment because delayed scaling requires persistent state for
-activations, and there isn't a clean and sound way to implement this with tensor subclasses.
-
-## single GPU
-
-### separation of concerns
-
-1. `Float8Linear` owns casting X, W and dL/dY to float8 and does all the bookkeeping of the amax, amax_history and scale buffers
-2. user is responsible for applying `Float8Linear` to the right parts of their model with module swaps
-
-
-### Tensor subclasses
-
-We are using tensor subclasses (`Float8Tensor`) to write modular code which satisfies
-autograd's restriction that `x.dtype == x.grad.dtype`.  The way we achieve this is by
-ensuring that instances of `Float8Tensor` set their dtype attribute to the original
-dtype (float32/float16/bfloat16) while the underlying data representation is in float8.
-If you look in `float8_linear.py` and `te_linear.py`, you will see that we pass instances of `Float8Tensor`
-around various `torch.autograd.Function` calls, enabling us to have modular code.
-
-## multi GPU
-
-### TP/SP
```

Comment:
Maybe we still have the TP/SP stuff documented just saying we know this doesn't work with compile and planto have dtensor integration for this

Reply:
hmm, it's not that usable without compile though :(  I'm more excited about just deleting this